wandb/sdk/launch/sweeps/launch_sweep.py

```diff
@@ -0,0 +1,294 @@
+import json
+from typing import Any, Dict, List, Optional, Union
+
+import click
+import yaml
+
+import wandb
+from wandb.apis import InternalApi, PublicApi
+from wandb.sdk.launch import utils as launch_utils
+from wandb.sdk.launch.sweeps.scheduler import Scheduler
+from wandb.sdk.launch.sweeps.utils import (
+    check_job_exists,
+    construct_scheduler_args,
+    get_previous_args,
+    handle_sweep_config_violations,
+)
+
+
+def launch_sweep(
+    entity: str,
+    project: str,
+    queue: str,
+    parsed_user_config: Optional[Dict[str, Any]] = None,
+    resume_id: Optional[str] = None,
+) -> Optional[str]:
+    """Create a sweep and launch it on a remote resource queue.
+
+    Arguments:
+        entity (str): The entity to launch the sweep under
+        project (str): The project to launch the sweep under
+        queue (str): The queue name to launch the sweep to
+        parsed_user_config (Optional[Dict[str, Any]]): a dictionary of sweep and launch config parameters
+        resume_id (Optional[str]): The id of the sweep to resume
+
+    Returns:
+        str: The sweep id of the launched sweep
+
+    Example:
+        config = {
+            "job": "wandb/sweep-jobs/job-fashion-MNIST-train:latest",
+            "method": "grid",
+            "run_cap": 2,
+            "parameters": {
+                "learning_rate": {
+                    "values": [0.01, 0.1]
+                },
+                "batch_size": {
+                    "values": [32, 64]
+                },
+            },
+            "scheduler": {
+                "num_workers": 2,
+                "resource": "local-container"
+            },
+            "launch": {
+                "resource_args": {
+                    "local-container": {
+                        "env": ["WANDB_BASE_URL", "WANDB_SILENT"]
+                    }
+                }
+            }
+        }
+    """
+    api = InternalApi()
```

Comment:
it gets really hard to read from here on down.  are there significant changes in the code or is this just a straight extraction?

Reply:
Functionally an extraction, with slight re-ordering to improve logical flow. I'll put some comments in there.